lib/puppet/type/rabbitmq_plugin.rb

```diff
@@ -26,6 +32,15 @@
     newvalues(%r{^\S+$})
   end
 
+  newparam(:mode) do
```

Comment:
`mode` seems a little potentially ambiguous with file modes, but also, I think an `offline` Boolean would probably be cleaner?

Reply:
First I used a bool but why offline => true and not online => false
It's not like the flag durable which should be true or false.
Then I used a free param "options" where I just put "--offline".
Finally I found that specify offline or online explicitly is easier. 

If you think that it's more clean and simple for a user, why not :)